pkg/virt-handler/vm_test.go

```diff
@@ -2193,36 +2194,23 @@ var _ = Describe("VirtualMachineInstance", func() {
 			Expect(blockMigrate).To(BeTrue())
 			Expect(err).To(Equal(fmt.Errorf("cannot migrate VMI with non-shared HostDisk")))
 		})
-		table.DescribeTable("when host model labels", func(toDefineHostModelLabels bool) {
+		table.DescribeTable("when host model labels", func(capabilities *nodelabellerapi.Capabilities) {
 			vmi := api2.NewMinimalVMI("testvmi")
 			vmi.Spec.Domain.CPU = &v1.CPU{Model: v1.CPUModeHostModel}
 
-			node := &k8sv1.Node{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:   host,
-					Labels: map[string]string{},
-				},
-			}
-
-			if toDefineHostModelLabels {
-				node.ObjectMeta.Labels = map[string]string{
-					v1.HostModelCPULabel + "fake":              "true",
-					v1.HostModelRequiredFeaturesLabel + "fake": "true",
-				}
-			}
-			addNode(clientTest, node)
+			controller.capabilities = capabilities
 
 			err := controller.isHostModelMigratable(vmi)
 
-			if toDefineHostModelLabels {
+			if capabilities != nil {
 				Expect(err).ShouldNot(HaveOccurred())
 			} else {
 				Expect(err).Should(HaveOccurred())
 			}
 
 		},
-			table.Entry("exist migration should succeed", true),
-			table.Entry("don't exist migration should fail", false),
+			table.Entry("exist migration should succeed", &nodelabellerapi.Capabilities{Host: nodelabellerapi.Host{UUID: "hostUUID"}}),
```

Comment:
This proves my point perfectly.

In this test you provide capabilities that does not have any details about the host's CPU (a bad state) and yet the test passes without a problem.

I think you should start by expecting this test to fail, then create another one with proper capabilities and expect it to pass.